core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -818,10 +822,13 @@ protected Scope visitCreateTableAsSelect(CreateTableAsSelect node, Optional<Scop
             // create target table metadata
             CatalogName catalogName = getRequiredCatalogHandle(metadata, session, node, targetTable.getCatalogName());
 
+            if (hasDefaultProperty(node.getProperties())) {
```

Comment:
ditto: move property support to `getProperties`

Reply:
Do we want to allow a user to set a property to `DEFAULT` in `CTAS`?